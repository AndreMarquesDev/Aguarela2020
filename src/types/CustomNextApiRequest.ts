import type { NextApiRequest } from 'next';
import type { FormPostRequestBody } from './FormPostRequestBody';

export type CustomNextApiRequest = {
    body: FormPostRequestBody;
} & NextApiRequest;
