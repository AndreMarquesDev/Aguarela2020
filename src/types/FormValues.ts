export type FormValues = {
    name: string;
    email: string;
    message: string;
    brand: string;
    subject: string;
};
