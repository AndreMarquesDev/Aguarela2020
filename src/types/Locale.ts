export enum Locale {
    Pt = 'pt',
    En = 'en',
}
