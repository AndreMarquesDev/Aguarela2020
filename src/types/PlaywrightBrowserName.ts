export enum PlaywrightBrowserName {
    Chromium = 'chromium',
    Firefox = 'firefox',
    Webkit = 'webkit',
}
