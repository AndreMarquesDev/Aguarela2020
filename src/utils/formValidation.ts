export enum FieldTypes {
    Name = 'name',
    Email = 'email',
    Textarea = 'message',
    Brand = 'brand',
    Subject = 'subject',
}
export type FormState = {
    name: string;
    email: string;
    message: string;
    brand: string;
    subject: string;
};

export const validateName = (value: string): boolean => {
    const nameRegex =
        // eslint-disable-next-line regexp/no-unused-capturing-group, regexp/prefer-plus-quantifier, regexp/prefer-character-class, regexp/no-super-linear-backtracking, regexp/no-useless-escape
        /s*([a-zA-ZáâàãéêèíîìóôòõúûùñçäöüßÁÂÀÃÉÊÈÍÎÌÓÔÒÕÚÛÙÑÇÄÖÜẞ]{1,}([\.,] |[-']| ))+[a-zA-ZáâàãéêèíîìóôòõúûùñçäöüßÁÂÀÃÉÊÈÍÎÌÓÔÒÕÚÛÙÑÇÄÖÜẞ]+\.?\s*$/;

    return nameRegex.test(value);
};

export const validateEmail = (value: string): boolean => {
    const emailRegex =
        // eslint-disable-next-line regexp/no-useless-escape, regexp/no-unused-capturing-group, regexp/prefer-d, regexp/use-ignore-case
        /(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;

    return emailRegex.test(value);
};
