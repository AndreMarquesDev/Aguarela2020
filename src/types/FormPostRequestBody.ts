import { FormValues } from './FormValues';

export interface FormPostRequestBody extends FormValues {
    isPlaywright: boolean;
}
