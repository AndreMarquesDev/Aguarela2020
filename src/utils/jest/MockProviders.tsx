/* eslint-disable react/jsx-no-constructed-context-values */
import React, { ReactNode } from 'react';
import { NavLinksContext } from '../../components/context/NavLinksContext';
import { TextsContext } from '../../components/context/TextsContext';
import { Locale } from '../../types/Locale';
import { textsEn, textsPt } from '../texts';

interface MockProvidersProps {
    language?: Locale;
    isMenuOpen?: boolean;
    toggleMenu?: jest.Mock;
    setNavHeight?: jest.Mock;
    children: ReactNode;
}

export const MockProviders = ({
    language = Locale.Pt,
    isMenuOpen = false,
    toggleMenu = jest.fn(),
    setNavHeight = undefined,
    children,
}: MockProvidersProps): JSX.Element => {
    return (
        <TextsContext.Provider
            value={{
                texts: language === Locale.Pt ? textsPt : textsEn,
            }}
        >
            <NavLinksContext.Provider value={{ isMenuOpen, toggleMenu, setNavHeight }}>
                {children}
            </NavLinksContext.Provider>
        </TextsContext.Provider>
    );
};
