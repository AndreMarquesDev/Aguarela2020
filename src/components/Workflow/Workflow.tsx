import React, { useContext } from 'react';
import { workflowSectionDataTestId } from '../../utils/dataTestIds';
import { TextsContext } from '../context/TextsContext';
import { Title } from '../Title/Title';
import {
    mobileMediaQuery,
    phabletMediaQuery,
    tabletMediaQuery,
    wrapperMediaQuery,
} from '../../styles/mediaQueries';

export const Workflow = (): JSX.Element => {
    const { texts } = useContext(TextsContext);

    return (
        <>
            <section className="container" data-testid={workflowSectionDataTestId}>
                <div className="wrapper genericMargins">
                    <Title text={texts.workflow} />

                    <ul>
                        <li className="fontXS">
                            <p>1.</p>
                            <p>{texts.defineTarget}</p>
                            <div className="backface">
                                <p>{texts.whatIsTheTarget}</p>
                            </div>
                        </li>
                        <li className="fontXS">
                            <p>2.</p>
                            <p>{texts.defineGoals}</p>
                            <div className="backface">
                                <p>{texts.whatAreYourGoals}</p>
                            </div>
                        </li>
                        <li className="fontXS">
                            <p>3.</p>
                            <p>{texts.platformStrategy}</p>
                            <div className="backface">
                                <p>{texts.whatPlatforms}</p>
                            </div>
                        </li>
                        <li className="fontXS">
                            <p>4.</p>
                            <p>{texts.contentCreation}</p>
                            <div className="backface">
                                <p>{texts.relevantContentCreation}</p>
                            </div>
                        </li>
                        <li className="fontXS">
                            <p>5.</p>
                            <p>{texts.paidSocialAndInfluencers}</p>
                            <div className="backface">
                                <p>{texts.promotePostsAndCollaborate}</p>
                            </div>
                        </li>
                        <li className="fontXS">
                            <p>6.</p>
                            <p>{texts.insightsAndReports}</p>
                            <div className="backface">
                                <p>{texts.detailedAnalysis}</p>
                            </div>
                        </li>
                        <li className="fontXS">
                            <p>7.</p>
                            <p>{texts.optimization}</p>
                            <div className="backface">
                                <p>{texts.improvementsAndUpdatesToTheStrategy}</p>
                            </div>
                        </li>
                    </ul>
                </div>
            </section>

            <style jsx>
                {`
                    .container {
                        --blockSize: 160rem;
                        --blockSizeMobile: 120rem;
                        --backfaceOffset: 12rem;

                        background: var(--purple);

                        ul {
                            display: flex;
                            justify-content: space-between;

                            @media (${wrapperMediaQuery}) {
                                width: 800rem;
                                flex-wrap: wrap;
                                justify-content: center;
                                margin: 0 auto;
                            }

                            @media (${phabletMediaQuery}) {
                                width: 600rem;
                            }

                            @media (${tabletMediaQuery}) {
                                width: 400rem;
                            }

                            @media (${mobileMediaQuery}) {
                                width: 280rem;
                                justify-content: space-between;
                            }

                            li {
                                width: var(--blockSize);
                                height: var(--blockSize);
                                display: flex;
                                flex-direction: column;
                                justify-content: center;
                                position: relative;
                                text-align: center;
                                padding: 10rem;
                                z-index: 0;

                                @media (${wrapperMediaQuery}) {
                                    margin: 0 calc(7rem + var(--backfaceOffset)) 0 7rem;

                                    &:first-child {
                                        margin-bottom: 50rem;
                                    }
                                }

                                @media (${phabletMediaQuery}) {
                                    &:nth-child(4) {
                                        margin-bottom: 50rem;
                                    }
                                }

                                @media (${tabletMediaQuery}) {
                                    &:nth-child(5) {
                                        margin-bottom: 50rem;
                                    }
                                }

                                @media (${mobileMediaQuery}) {
                                    width: var(--blockSizeMobile);
                                    height: var(--blockSizeMobile);
                                    margin: 0;
                                }

                                &:before,
                                &:after {
                                    content: '';
                                    width: var(--blockSize);
                                    height: var(--blockSize);
                                    position: absolute;

                                    @media (${mobileMediaQuery}) {
                                        width: var(--blockSizeMobile);
                                        height: var(--blockSizeMobile);
                                    }
                                }

                                &:before {
                                    top: 0;
                                    left: 0;
                                    background-color: var(--blue);
                                }

                                &:after {
                                    top: var(--backfaceOffset);
                                    left: var(--backfaceOffset);
                                    background-color: var(--pink);
                                    z-index: -1;
                                }

                                &:hover {
                                    .backface {
                                        display: flex;
                                    }

                                    &:after {
                                        background-color: var(--blue);
                                    }
                                }

                                p {
                                    color: var(--yellow);
                                    font-weight: bold;
                                    text-transform: uppercase;
                                    position: relative;
                                    display: block;
                                    z-index: 0;

                                    @media (${mobileMediaQuery}) {
                                        font-size: var(--fontSizeXXS);
                                    }
                                }

                                .backface {
                                    display: none;
                                    width: 100%;
                                    height: 100%;
                                    position: absolute;
                                    top: 0;
                                    left: 50%;
                                    transform: translateX(-50%);
                                    align-items: center;
                                    background: var(--pink);
                                    padding: 10rem;

                                    p {
                                        color: var(--yellow);
                                        font-weight: normal;
                                        text-transform: none;

                                        @media (${mobileMediaQuery}) {
                                            font-size: var(--fontSizeXXS);
                                        }
                                    }
                                }
                            }
                        }
                    }
                `}
            </style>
        </>
    );
};
