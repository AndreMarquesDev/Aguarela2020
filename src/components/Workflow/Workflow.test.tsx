import type { RenderResult } from '@testing-library/react';
import { render, screen } from '@testing-library/react';
import React from 'react';
import { Locale } from '../../types/Locale';
import { MockProviders } from '../../utils/jest/MockProviders';
import { textsEn, textsPt } from '../../utils/texts';
import { Workflow } from './Workflow';
import '@testing-library/jest-dom';

const renderComponent = (language: Locale = Locale.Pt): RenderResult => {
    return render(
        <MockProviders language={language}>
            <Workflow />
        </MockProviders>,
    );
};

describe('<Workflow />', () => {
    test('renders properly', () => {
        const { container } = renderComponent();

        expect(screen.getByText(textsPt.workflow)).toBeInTheDocument();
        expect(screen.getByText(textsPt.defineTarget)).toBeInTheDocument();
        expect(screen.getByText(textsPt.whatIsTheTarget)).toBeInTheDocument();
        expect(screen.getByText(textsPt.defineGoals)).toBeInTheDocument();
        expect(screen.getByText(textsPt.whatAreYourGoals)).toBeInTheDocument();
        expect(screen.getByText(textsPt.platformStrategy)).toBeInTheDocument();
        expect(screen.getByText(textsPt.whatPlatforms)).toBeInTheDocument();
        expect(screen.getByText(textsPt.contentCreation)).toBeInTheDocument();
        expect(screen.getByText(textsPt.relevantContentCreation)).toBeInTheDocument();
        expect(screen.getByText(textsPt.paidSocialAndInfluencers)).toBeInTheDocument();
        expect(screen.getByText(textsPt.promotePostsAndCollaborate)).toBeInTheDocument();
        expect(screen.getByText(textsPt.insightsAndReports)).toBeInTheDocument();
        expect(screen.getByText(textsPt.detailedAnalysis)).toBeInTheDocument();
        expect(screen.getByText(textsPt.optimization)).toBeInTheDocument();
        expect(screen.getByText(textsPt.improvementsAndUpdatesToTheStrategy)).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });

    test('renders properly in English', () => {
        const { container } = renderComponent(Locale.En);

        expect(screen.getByText(textsEn.workflow)).toBeInTheDocument();
        expect(screen.getByText(textsEn.defineTarget)).toBeInTheDocument();
        expect(screen.getByText(textsEn.whatIsTheTarget)).toBeInTheDocument();
        expect(screen.getByText(textsEn.defineGoals)).toBeInTheDocument();
        expect(screen.getByText(textsEn.whatAreYourGoals)).toBeInTheDocument();
        expect(screen.getByText(textsEn.platformStrategy)).toBeInTheDocument();
        expect(screen.getByText(textsEn.whatPlatforms)).toBeInTheDocument();
        expect(screen.getByText(textsEn.contentCreation)).toBeInTheDocument();
        expect(screen.getByText(textsEn.relevantContentCreation)).toBeInTheDocument();
        expect(screen.getByText(textsEn.paidSocialAndInfluencers)).toBeInTheDocument();
        expect(screen.getByText(textsEn.promotePostsAndCollaborate)).toBeInTheDocument();
        expect(screen.getByText(textsEn.insightsAndReports)).toBeInTheDocument();
        expect(screen.getByText(textsEn.detailedAnalysis)).toBeInTheDocument();
        expect(screen.getByText(textsEn.optimization)).toBeInTheDocument();
        expect(screen.getByText(textsEn.improvementsAndUpdatesToTheStrategy)).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });
});
