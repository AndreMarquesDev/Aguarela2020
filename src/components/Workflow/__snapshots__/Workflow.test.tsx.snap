// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Workflow /> renders properly 1`] = `
<div>
  <section
    class="jsx-2192258247 container"
    data-testid="workflow_section"
  >
    <div
      class="jsx-2192258247 wrapper genericMargins"
    >
      <h1
        class="jsx-3953086084 "
      >
        Workflow
      </h1>
      <ul
        class="jsx-2192258247"
      >
        <li
          class="jsx-2192258247"
        >
          <p
            class="jsx-2192258247"
          >
            1.
          </p>
          <p
            class="jsx-2192258247"
          >
            Definir target
          </p>
          <div
            class="jsx-2192258247 backface"
          >
            <p
              class="jsx-2192258247"
            >
              Quem é o público-alvo da marca
            </p>
          </div>
        </li>
        <li
          class="jsx-2192258247"
        >
          <p
            class="jsx-2192258247"
          >
            2.
          </p>
          <p
            class="jsx-2192258247"
          >
            Definir objetivos
          </p>
          <div
            class="jsx-2192258247 backface"
          >
            <p
              class="jsx-2192258247"
            >
              Quais são os objetivos de negócio e comunicação
            </p>
          </div>
        </li>
        <li
          class="jsx-2192258247"
        >
          <p
            class="jsx-2192258247"
          >
            3.
          </p>
          <p
            class="jsx-2192258247"
          >
            Estratégia de plataformas
          </p>
          <div
            class="jsx-2192258247 backface"
          >
            <p
              class="jsx-2192258247"
            >
              Quais as plataformas onde a marca deve estar presente
            </p>
          </div>
        </li>
        <li
          class="jsx-2192258247"
        >
          <p
            class="jsx-2192258247"
          >
            4.
          </p>
          <p
            class="jsx-2192258247"
          >
            Criação de Conteúdo
          </p>
          <div
            class="jsx-2192258247 backface"
          >
            <p
              class="jsx-2192258247"
            >
              Criação de conteúdo relevante e de qualidade para o target
            </p>
          </div>
        </li>
        <li
          class="jsx-2192258247"
        >
          <p
            class="jsx-2192258247"
          >
            5.
          </p>
          <p
            class="jsx-2192258247"
          >
            Paid social & influencers
          </p>
          <div
            class="jsx-2192258247 backface"
          >
            <p
              class="jsx-2192258247"
            >
              Promoção de posts e colaborações com influencers
            </p>
          </div>
        </li>
        <li
          class="jsx-2192258247"
        >
          <p
            class="jsx-2192258247"
          >
            6.
          </p>
          <p
            class="jsx-2192258247"
          >
            Insights e reports
          </p>
          <div
            class="jsx-2192258247 backface"
          >
            <p
              class="jsx-2192258247"
            >
              Análises detalhadas e identificação de insights
            </p>
          </div>
        </li>
        <li
          class="jsx-2192258247"
        >
          <p
            class="jsx-2192258247"
          >
            7.
          </p>
          <p
            class="jsx-2192258247"
          >
            Otimização
          </p>
          <div
            class="jsx-2192258247 backface"
          >
            <p
              class="jsx-2192258247"
            >
              Melhorias e updates à estratégia
            </p>
          </div>
        </li>
      </ul>
    </div>
  </section>
</div>
`;

exports[`<Workflow /> renders properly in English 1`] = `
<div>
  <section
    class="jsx-2192258247 container"
    data-testid="workflow_section"
  >
    <div
      class="jsx-2192258247 wrapper genericMargins"
    >
      <h1
        class="jsx-3953086084 "
      >
        Workflow
      </h1>
      <ul
        class="jsx-2192258247"
      >
        <li
          class="jsx-2192258247"
        >
          <p
            class="jsx-2192258247"
          >
            1.
          </p>
          <p
            class="jsx-2192258247"
          >
            Set the target
          </p>
          <div
            class="jsx-2192258247 backface"
          >
            <p
              class="jsx-2192258247"
            >
              What's the target
            </p>
          </div>
        </li>
        <li
          class="jsx-2192258247"
        >
          <p
            class="jsx-2192258247"
          >
            2.
          </p>
          <p
            class="jsx-2192258247"
          >
            Set goals
          </p>
          <div
            class="jsx-2192258247 backface"
          >
            <p
              class="jsx-2192258247"
            >
              What are the business and communication goals
            </p>
          </div>
        </li>
        <li
          class="jsx-2192258247"
        >
          <p
            class="jsx-2192258247"
          >
            3.
          </p>
          <p
            class="jsx-2192258247"
          >
            Platforms strategies
          </p>
          <div
            class="jsx-2192258247 backface"
          >
            <p
              class="jsx-2192258247"
            >
              In what platforms should the brand be present
            </p>
          </div>
        </li>
        <li
          class="jsx-2192258247"
        >
          <p
            class="jsx-2192258247"
          >
            4.
          </p>
          <p
            class="jsx-2192258247"
          >
            Content creation
          </p>
          <div
            class="jsx-2192258247 backface"
          >
            <p
              class="jsx-2192258247"
            >
              Relevant and quality content created for the target
            </p>
          </div>
        </li>
        <li
          class="jsx-2192258247"
        >
          <p
            class="jsx-2192258247"
          >
            5.
          </p>
          <p
            class="jsx-2192258247"
          >
            Paid social & influencers
          </p>
          <div
            class="jsx-2192258247 backface"
          >
            <p
              class="jsx-2192258247"
            >
              Promoted posts and collaborations with influencers
            </p>
          </div>
        </li>
        <li
          class="jsx-2192258247"
        >
          <p
            class="jsx-2192258247"
          >
            6.
          </p>
          <p
            class="jsx-2192258247"
          >
            Insights and reports
          </p>
          <div
            class="jsx-2192258247 backface"
          >
            <p
              class="jsx-2192258247"
            >
              Detailed analysis and insights' finding
            </p>
          </div>
        </li>
        <li
          class="jsx-2192258247"
        >
          <p
            class="jsx-2192258247"
          >
            7.
          </p>
          <p
            class="jsx-2192258247"
          >
            Optimization
          </p>
          <div
            class="jsx-2192258247 backface"
          >
            <p
              class="jsx-2192258247"
            >
              Strategy updates and improvements
            </p>
          </div>
        </li>
      </ul>
    </div>
  </section>
</div>
`;
